pkg/storegateway/bucket.go

```diff
@@ -1274,7 +1274,7 @@ func (s *BucketStore) openBlocksForReading(ctx context.Context, skipChunks bool,
 
 	indexReaders := make(map[ulid.ULID]*bucketIndexReader, len(blocks))
 	for _, b := range blocks {
-		indexReaders[b.meta.ULID] = b.loadedIndexReader(s.postingsStrategy, stats)
+		indexReaders[b.meta.ULID] = b.loadedIndexReader(spanCtx, s.postingsStrategy, stats)
```

Comment:
nit, but I'd add a comment on loadedIndexReader saying that the context is not retained after loadedIndexReader returns

Reply:
Good idea, added in d04e880.